source/graphql/migrate-hasura.txt

```diff
@@ -0,0 +1,263 @@
+Explorer page on the Hasura Console to test out some GraphQL queries.
+
+Hasura also uses the GraphiQL interface, which is similar to how you test
+queries in Atlas App Services.
+
+.. image:: /images/migration/test-queries.png
+   :alt: Use the GraphiQL interface to test queries.
+
+Authorization and Authentication
+--------------------------------
+
+Hasura does not directly handle authentication. Instead, it relies on session
+variables provided by an external authentication service. These session
+variables include user, role, and organization information crucial for
+determining data access rights. For details, refer to the `Hasura authentication
+docs <https://hasura.io/docs/latest/auth/overview/>`_
+
+Data access permissions, including roles and rule expressions, can be converted
+into Hasura role-based permission rules.
+
+All Authentication methods that Atlas provides are compatible with Hasura's
+Webhook and JWT auth methods. If you are using Email/Pass, Anonymous, or API Key
+authentication, use Hasura Webhook. If you are using a Custom JWT, integrate
+directly with Hasura's JWT auth method.
+
+It is recommended to use an external IdP for managing authentication
+processes. For enhanced security and flexibility, organizations typically manage
+tokens and keys centrally rather than adjacent to the database.
+
+You can integrate Hasura with any authentication provider of your choice, such
+as Auth0, Firebase Auth, AWS Cognito, or even a custom solution, to verify the
+user and set the necessary session variables. For configuring JWT or webhook
+authentication in Hasura, refer to the documentation at:
+
+- `JWT Authentication <https://hasura.io/docs/latest/auth/authentication/jwt/>`_
+- `Webhook Authentication <https://hasura.io/docs/latest/auth/authentication/webhook/>`_
+
+Set Up Custom Resolvers
+-----------------------
+
+If your existing GraphQL API endpoints include custom resolvers or business
+logic, you'll need to implement these in Hasura. Hasura supports custom business
+logic using remote schemas, event triggers, and actions.
+
+Hasura Actions can be used to create custom resolvers. Custom queries and
+mutations including their payload type and input type can be defined within
+Hasura, and Hasura can integrate with the resolver function over HTTP. 
+
+.. image:: /images/migration/actions.png
+   :alt: Use Hasura Actions to create custom resolvers.
+
+The resolver function must be deployed somewhere and Hasura can generate the
+JavaScript code for the function.
+
+.. image:: /images/migration/resolver-codegen.png
+   :alt: Hasura auto generates Action code.
+
+Update Client Applications
+--------------------------
+
+Update any client applications that interact with your GraphQL API endpoints to
+point to the new Hasura endpoint URLs. Any existing Apollo client will work with
+Hasura.
+
+Update the GraphQL operations used by applications to match Hasura-style GraphQL
+operations. 
+
+Shut down MongoDB Atlas App Services
```

Comment:
Can we also change this heading to "Shut down Atlas App Services Endpoints" instead of all of App Services? Users could have other apps with other services running on App Services, so don't want to make it sounds like we're deprecating all of App Services.

Reply:
Good point! I've updated.